rotkehlchen/chain/evm/transactions.py

```diff
@@ -447,6 +460,72 @@ def _get_erc20_transfers_for_ranges(
                 queried_ranges=[(start_ts, end_ts)],
             )
 
+    def address_has_been_spammed(self, address: ChecksumEvmAddress) -> bool:
+        """
+        Queries erc20 tranfers for the given address and if it has only spam assets or ignored
+        assets we return True.
+        If any transfer had an unknown asset
+        """
+        start_ts, end_ts = Timestamp(0), ts_now()
+        checekd_tokens = set()
+        with self.database.conn.read_ctx() as cursor:
+            ignored_assets = self.database.get_ignored_asset_ids(cursor)
+
+        log.debug(f'Address detection: querying {self.evm_inquirer.chain_name} ERC20 Transfers for {address} -> {start_ts} - {end_ts}')  # noqa: E501
+        try:
+            for erc20_tx_hashes in self.evm_inquirer.etherscan.get_token_transaction_hashes(
+                account=address,
+                from_ts=start_ts,
+                to_ts=end_ts,
+            ):
+                for tx_hash in erc20_tx_hashes:
+                    tx_hash_bytes = deserialize_evm_tx_hash(tx_hash)
+                    _, raw_receipt_data = self.evm_inquirer.get_transaction_by_hash(tx_hash_bytes)  # noqa: E501
```

Comment:
This gets both the transaction hash and the receipt data. You only need the receipts here. So just ask for the receipts with the appropriate function.